no-unused-vars is not reporting for object destructuring

No error:
```js
const {connect} = require('redux/react')
```

Error
`error  no-unused-vars  connect is defined but never used`
```
const connect = require('redux/react').connect
```

Confirmed.

app/utils.js :
```
const {Bacona} = require('baconjs')
```
---
` //"parser": "babel-eslint",`
```
$ ./node_modules/.bin/eslint app/utils.js 

app/utils.js
  2:7  error  Bacona is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

---
`"parser": "babel-eslint",`
```
$ ./node_modules/.bin/eslint app/utils.js 
$ 
```

--- a/acorn-to-esprima.js
+++ b/acorn-to-esprima.js
@@ -172,7 +172,7 @@ var astTransformVisitor = {
   noScope: true,
   exit: function (node) { /* parent */
     if (this.isSpreadProperty()) {
-      node.type = "Property";
+      node.type = "SpreadProperty";
       node.kind = "init";
       node.computed = true;
       node.key = node.value = node.argument;

--- a/index.js
+++ b/index.js
@@ -264,8 +264,15 @@ function monkeypatch() {
           checkIdentifierOrVisit.call(this, typeAnnotation);
         }
         if (id.type === "ObjectPattern") {
-          for (var j = 0; j < id.properties.length; j++) {
-            this.visit(id.properties[j]);
+          // check if object destructuring has a spread
+          var hasSpread = id.properties.filter(function(p) {
+            return p.type === "SpreadProperty"
+          });
+          // visit properties if so
+          if (hasSpread.length > 0) {
+            for (var j = 0; j < id.properties.length; j++) {
+              this.visit(id.properties[j]);
+            }
           }
         }
       }

--- a/test/non-regression.js
+++ b/test/non-regression.js
@@ -1171,4 +1171,12 @@ describe("verify", function () {
       []
     );
   });
+
+  it("detects no-unused-vars with object destructuring #142", function () {
+    verifyAndAssertMessages(
+      "const {Bacona} = require('baconjs')",
+      { "no-undef": 1, "no-unused-vars": 1 },
+      [ "1:7 Bacona is defined but never used no-unused-vars" ]
+    );
+  });
 });

